### PR TITLE
Allow ConfigureServices to be called multiple times.

### PR DESCRIPTION
**Pull request by Tratcher**
@davidfowl @JunTaoLuo 

Required for https://github.com/aspnet/IISIntegration/pull/97

### CODE REVIEW

File: src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs [hunk 1]
```diff
@@ -32,7 +33,7 @@ public class WebHostBuilder : IWebHostBuilder
         private IConfiguration _config = new ConfigurationBuilder().AddInMemoryCollection().Build();
         private WebHostOptions _options;
 
-        private Action<IServiceCollection> _configureServices;
+        private IList<Action<IServiceCollection>> _configureServicesDelegates = new List<Action<IServiceCollection>>();
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs]: nit: readonly and make it a list.

File: test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs [hunk 2]
```diff
@@ -149,6 +149,27 @@ public void UseCaptureStartupErrorsHonored()
...
+                })
+                .Configure(app => { });
+
+            var host = hostBuilder.Build();
+
+            Assert.NotNull(host.Services.GetRequiredService<ServiceA>());
+            Assert.NotNull(host.Services.GetRequiredService<ServiceB>());
+        }
+
+        [Fact]
```
Comment (JunTaoLuo) [inline test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs]: Should we add a test to check for the order that the ConfigureServices are called?

Comment (Tratcher) [inline test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs]: Added

**Comment by JunTaoLuo** [comment]
:shipit:

**Comment by davidfowl** [comment]
:shipit:

**Pull request merged**